test/parallel/test-buffer-alloc.js

```diff
@@ -75,16 +75,40 @@ new Buffer('', 'binary');
 Buffer(0);
 
 // try to write a 0-length string beyond the end of b
-assert.throws(() => b.write('', 2048), RangeError);
+common.expectsError(
+  () => b.write('', 2048),
+  {
```

Comment:
nit: Since this error object is used multiple times in the test, it can be stored in a const.